### ISSUE TITLE
FYI - Newest jspdf screws things up (1.4.1 -> 1.5.3)

### ISSUE THREAD

**Issue by rconstantine** [eKoopmans/html2pdf.js#217]
When I installed html2pdf.js in a new project, it grabbed the latest jspdf as its dependency. Well, my previously perfectly breaking pages suddenly a) didn't break right and b) didn't fit width-wise anymore. Rolling back to 1.4.1 fixed that.

This is being used in a Quasar (VueJS) project.

Here is my onPrint function that generates the PDF (notes follow):

```javascript
onPrint () {
      // DON'T UPGRADE JSPDF past V1.4.1
      this.$q.loading.show({
        spinner: 'QSpinnerGears',
        message: 'Generating PDF...',
        customClass: 'bg-blue-10'
      })
      let internalLoading = this.$q.loading
      let element = document.getElementById('reportComponent')
      let facilityName = this.facilityName
      let reportName = this.selected.name
      let locationAbbr = this.facility
      let today = moment()
      let opt = {
        margin: [100, 15, 75, 15],
        filename: this.selected.componentName + '-' + locationAbbr.value + ' - ' + today.format('MM/DD/YY HH:mm') + '.pdf',
        image: { type: 'jpeg', quality: 1 },
        // pagebreak: { mode: ['avoid-all', 'css'] },
        pagebreak: { avoid: '.avoid' },
        html2canvas: { scale: 2 },
        jsPDF: { unit: 'px', format: [925, 1197], orientation: 'portrait', compress: true, putOnlyUsedFonts: true }
      }
      this.$html2pdf().set(opt).from(element).toPdf().get('pdf').then(function (pdf) {
        // Your code to alter the pdf object.
        let totalPages = pdf.internal.getNumberOfPages()

        for (let i = 1; i <= totalPages; i++) {
          pdf.setPage(i)
          pdf.setFontSize(20)
          pdf.setTextColor(0)
          // header
          let img = new Image()
          img.src = '../../assets/SH Logo Blue.png'
          pdf.addImage('data:image/png;base64,SOMETHING HERE', 'PNG', 30, 20, 75, 75)
          pdf.text([facilityName], 450, 60, 'center')
          pdf.setFontStyle('bold')
          pdf.text([reportName + ' - ' + locationAbbr.value], 450, 80, 'center')

          // footer
          pdf.setFontStyle('normal')
          pdf.text('Printed: ' + today.format('ddd, MM/DD/YYYY HH:mm'), 35, 1167)
          pdf.text('Page ' + i + ' of ' + totalPages, 430, 1167)
          pdf.text('SutterNow Reports', 765, 1167)
        }
        internalLoading.hide()
      }).save()
      // console.log('worker', worker)
    }
```
My DIV that contains what I'm outputing is 900px wide. What happens with the newer jsPdf is that it doesn't fit in the 925 wide jsPdf document. The right side is cut off. It forces an increase to something like 1235 wide (and 1600 high), in order to show the whole contents, but the height doesn't work right. The page break is below the 1600 line and the header and footer will not work with the updated dimensions. In fact, I have to keep the header and footer as-is, even though the jsPdf width/height have changed. Additionally, the breaks happen in the middle of elements, or they're hidden below the bottom of the page. In other words, 'unbroken' elements aren't continued on to the next page.

I'm wondering if anyone else is seeing this. I think new folks might not know what they're seeing when this just doesn't work for them, but older users of html2pdf .js who upgrade jsPdf might be able to chime in with their results.

Also, this might not be related, but I can no longer install jsPdf 1.4.1 using npm. I had to just copy it from an older project. I was getting errors that didn't make sense and for which I couldn't find resolution online and didn't have time to investigate further.

**Comment by shuai4983958** [comment]
How is your width adapted to PDF

**Comment by eKoopmans** [comment]
Hi @rconstantine, thanks for the report and sorry I didn't respond earlier. I've had a few reports of issues with jsPDF v1.5.3, I haven't had a chance to investigate yet.

**Comment by eKoopmans** [comment]
This has been long overdue, sorry - should be fixed now in v0.9.3, which locks jsPDF to 1.4.1. I'll be looking at upgrading to jsPDF 2.x soon.
